simapp/sim_test.go

```diff
@@ -930,10 +930,7 @@ func TestAppImportExport(t *testing.T) {
 		storeB := ctxB.KVStore(storeKeyB)
 		kvA, kvB, count, equal := sdk.DiffKVStores(storeA, storeB, prefixes)
 		fmt.Printf("Compared %d key/value pairs between %s and %s\n", count, storeKeyA, storeKeyB)
-		require.True(t, equal,
-			"unequal stores: %s / %s:\nstore A %X => %X\nstore B %X => %X",
-			storeKeyA, storeKeyB, kvA.Key, kvA.Value, kvB.Key, kvB.Value,
-		)
+		require.True(t, equal, getSimulationLog(storeKeyA.Name(), app.cdc, newApp.cdc, kvA, kvB))
```

Comment:
Can you provide example output of what this looks like (when it fails) please?

Reply:
it doesn't fail, it'd just panic. I believe I covered all the cases on the tests to avoid that situation

Reply:
You misunderstand. I mean can you provide output of what this PR improves upon. Like what is the point of this PR? To improve the output? I'd like to see an example of that 👍

Reply:
oh yes, to improve the output of the simulation. 

Prev: 
```
--- FAIL: TestAppImportExport (2.87s)
    require.go:1159: 
                Error Trace:    sim_test.go:946
                Error:          Should be true
                Test:           TestAppImportExport
                Messages:       unequal stores: KVStoreKey{0xc00038a880, acc} / KVStoreKey{0xc0020ecd70, acc}:
                                store A 676C6F62616C4163636F756E744E756D626572 => 0123
                                store B 676C6F62616C4163636F756E744E756D626572 => 0117
```


Now:

```
--- FAIL: TestAppImportExport (2.53s)
    require.go:1159: 
                Error Trace:    sim_test.go:946
                Error:          Should be true
                Test:           TestAppImportExport
                Messages:       GlobalAccNumberA: 35
                                GlobalAccNumberB: 23
```

Reply:
here's an example from accounts: 

```

--- FAIL: TestAppImportExport (2.77s)
    require.go:1159: 
                Error Trace:    sim_test.go:946
                Error:          Should be true
                Test:           TestAppImportExport
                Messages:       |
                                  address: cosmos1mq3jmwvenzf2ax9psf0ruhjd3v3mt6cnqs0vmr
                                  coins: []
                                  pubkey: ""
                                  accountnumber: 23
                                  sequence: 0
                                  name: FeeCollector
                                  permission: basic
                            
                                |
                                  address: cosmos1mq3jmwvenzf2ax9psf0ruhjd3v3mt6cnqs0vmr
                                  coins: []
                                  pubkey: ""
                                  accountnumber: 22
                                  sequence: 0
                                  name: FeeCollector
                                  permission: basic
```